src/main/java/com/karankumar/bookproject/ui/goal/ReadingGoalView.java

```diff
@@ -247,42 +248,26 @@ String calculateProgress(int booksToReadThisYear, int booksReadThisYear) {
      * @param booksReadThisYear the number of books read so far
      * @return the number of books that the user is ahead or behind schedule by
      */
-    private int howFarAheadOrBehindSchedule(int booksToReadThisYear, int booksReadThisYear) {
-        int shouldHaveRead = booksToReadFromStartOfYear(booksToReadThisYear) * getWeekOfYear();
+    public int howFarAheadOrBehindSchedule(int booksToReadThisYear, int booksReadThisYear) {
+        int shouldHaveRead = booksToReadFromStartOfYear(booksToReadThisYear) * dateUtils.getWeekOfYear();
         return Math.abs(shouldHaveRead - booksReadThisYear);
     }
 
-    /**
-     * @return the current week number of the year
-     */
-    private int getWeekOfYear() {
-        LocalDate now = LocalDate.now();
-        WeekFields weekFields = WeekFields.of(Locale.getDefault());
-        return now.get(weekFields.weekOfWeekBasedYear());
-    }
-
-    /**
-     * @param weekOfYear the current week number of the year
-     * @return the number of weeks left in the year from the current week
-     */
-    private static int weeksLeftInYear(int weekOfYear) {
-        return (WEEKS_IN_YEAR - weekOfYear);
-    }
 
     /**
      * @param booksToReadThisYear the number of books to read by the end of the year (the goal)
      * @return the number of books that should have been read a week (on average) from the start of the year
      */
     public static int booksToReadFromStartOfYear(int booksToReadThisYear) {
-        return ((int) Math.ceil(booksToReadThisYear / WEEKS_IN_YEAR));
+        return ((int) Math.ceil(booksToReadThisYear / dateUtils.getWeeksInYear()));
     }
```

Comment:
`getWeeksInYear` can be invoked through static reference?

Reply:
I have changed the approach to remove the static keyword.